documentation/modules/exploit/windows/fileformat/theme_dll_hijack_cve_2023_38146.md

```diff
@@ -0,0 +1,121 @@
+## Description
+
+There exists a time of check to time of use vulnerability in the way Windows 11 loads msstyles files when they are
+via a theme file. When a user opens a theme which references an msstyles file with a `PACKME_VERSION`
+of 999, the process will check for the presence of the msstyles file appended with "_vrf.dll".  If the file is found,
+the process will open the file to check for a signature.  If the signature is valid, the process closes the file and
+then loads it. By closing the file after the check and before loading it, we can feed a legitimate signed dll to the
+check read, and then substitute a malicious dll for the second, resulting in the process loading our dll and executing
+arbitrary code.
+
+To achieve this race condition, we implement a UNC path pointing back to an SMB server we control that uses the type
+of request issued by the SMB client to decide to server the legitimate file or the payload file.
+
+Because the PACKME_VERSION must be 999, this module includes a tool to take a normal windows aero.msstyles file
+and populate it with the required PACKME_VERSION.  As the aero.msstyles file is also a signed binary, we can use it
+as both the msstyles file and the legitimate signed dll file.
```

Comment:
Maybe add something like `The "aero.msstyles" file can be found in C:\...` on a windows install.